client/src/components/RatingsAndReviews/AddReviewForm.jsx

```diff
@@ -0,0 +1,418 @@
+/* eslint-disable no-param-reassign */
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import AddCharacteristic from './AddCharacteristic';
+
+const AddReviewForm = class extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      chars: [],
+      showStarLabel: false,
+      showRatingWarning: false,
+      innerWidth: '0%',
+      rating: null,
+      summary: '',
+      body: '',
+      recommend: false,
+      name: '',
+      email: '',
+      photos: [],
+      characteristics: {},
+    };
+
+    this.processCharacteristics = this.processCharacteristics.bind(this);
+    this.handleStarRatingClick = this.handleStarRatingClick.bind(this);
+    this.handleCharRatingClick = this.handleCharRatingClick.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.fileInput = React.createRef();
+    this.handleFileUpload = this.handleFileUpload.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.processCharacteristics();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { characteristics } = this.props;
+    if (JSON.stringify(characteristics) !== JSON.stringify(prevProps.characteristics)) {
+      this.processCharacteristics();
+    }
+  }
+
+  handleStarRatingClick(id) {
+    this.setState({
+      innerWidth: String(id * 20).concat('%'),
+      showStarLabel: true,
+      rating: id,
+    });
+  }
+
+  handleCharRatingClick(id, e) {
+    const { characteristics } = this.state;
+    const selection = { ...characteristics };
+    selection[id] = Number(e.target.value);
+    this.setState({
+      characteristics: selection,
+    });
+    // console.log(e.target.value);
+    // console.log(e.target.name);
+  }
+
+  handleInputChange(e) {
+    const { name } = e.target;
+    this.setState({ [name]: e.target.value });
+  }
+
+  handleFileUpload(e) {
+    console.log(this.fileInput.current.files);
```

Comment:
a console log!

Reply:
But I see that it won't appear until the file upload functionality is complete --

Reply:
Yes, the button that triggers this function is not rendering currently. Saving this for future optimization and use of FileReader!